content/docs/1.2.4/volumes-and-nodes/volume-size.md

```diff
@@ -19,29 +19,90 @@ In this section, you'll have a better understanding of concepts related to volum
+    >
+    > Even if users will not take snapshots for volumes, there are operations like rebuilding, expansion, or backing up that would lead to system (hidden) snapshot creation. As a result, volume `actual size` being larger than size is unavoidable under some use cases.
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig5.png" >}}
+
+8. Delete snapshot#1 and wait for snapshot purge complete. See Figure 7 of the illustration.
     - Here Longhorn actually coalesces the snapshot#1 with the snapshot#2.
-    - For the overlapping part during the coalesce, the newer data (data#2) will be retained in the blocks. Then some historical data is removed and the volume gets shrunk (from 6.1Gi to 4.65Gi in the example).
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-4.png" >}}
+    - For the overlapping part during the coalesce, the newer data (data#2) will be retained in the blocks. Then some historical data is removed and the volume gets shrunk (from 16.3 Gi to 11.4 Gi in the example).
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig6.png" >}}
+
+Here we summarize the issues we encountered in the example:
+1. Unused blocks not releasing issue
+    Longhorn does not support TRIM/UNMAP operations. Hence deleting files from filesystems will not lead to volume actual size decreasing/shrinking.
+2. Allocated but unused blocks not reusing issue
+    Deleting then writing new files would lead to the actual size keeps increasing. Since the filesystem may not reuse the recently freed blocks from recently deleted files. Thus, allocating an appropriate nominal size for a volume that holds heavy writing tasks according to the IO pattern would make disk space usage more efficient.
+3. By deleting snapshots, the overlapping part of the used blocks might be eliminated regardless of whether the blocks are recently released blocks by the filesystem or still contain historical data.
+## Space Configuration Suggestions for Volumes
+
+1. Reserve enough free space in disks as buffers in case of the actual size of existing volumes keep growing up.
+    - A quick estimation for maximum space consumption estimation of a volume is
+  
+        ```
+        (N + 1 + 1) x head/snapshot average actual size
+        ```
+
+        where N is the number of snapshots retained, and the 1st 1 is for the system snapshot, and 2nd 1 is for the tmp space may required by snapshot deletion.
+    - If there are heavy writing tasks for volumes, head/snapshot average actual size would be volume the nominal size. In this case, it's better to set `Storage Over Provisioning Percentage` to be smaller than 100% to avoid disk space exhaustion.
```

Comment:
```suggestion
    - If there are heavy writing tasks for volumes, the head/snapshot average actual size would be volume the nominal size. In this case, it's better to set `Storage Over Provisioning Percentage` to be smaller than 100% to avoid disk space exhaustion.
```